eth/downloader/downloader.go

```diff
@@ -440,10 +442,18 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td *big.I
 	}
 	height := latest.Number.Uint64()
 
-	origin, err := d.findAncestor(p, latest)
-	if err != nil {
-		return err
+	var origin = uint64(0)
+
+	if d.blockchain != nil &&
+		d.blockchain.CurrentHeader() != nil &&
+		d.blockchain.CurrentHeader().Number != nil &&
+		d.blockchain.CurrentHeader().Number.Uint64() > 0 {
```

Comment:
This looks extremely defensive. Why did you add all these checks? 
It looks pretty moot anyway -- if `d.blockchain` is nul, you'll get a panic later on when this falls through to e.g `localHeight = d.blockchain.CurrentBlock().NumberU64`

Reply:
Thanks for pointing this out. Not one of my prouder moments :face_with_thermometer: . The original reasoning was that `les` tests failed otherwise, since `d.blockchain` would be nil. The middle two conditions I can only explain as results of frustrated debugging, and they are unnecessary.

What is currently unaddressed is that light mode is completely excluded from the condition, so common ancestor negotiation would  never be attempted during Light sync. Will push a change to fix this very shortly.

Reply:
See https://github.com/ethereum/go-ethereum/pull/21063/commits/4122416ba30e248d473bf5d67930168bc7608142